### PR TITLE
feat: add phone field

### PR DESCRIPTION
**Pull request by eugenederii**

### CODE REVIEW

File: src/modules/core/components/phone-field/phone-field.module.scss [hunk 1]
```diff
@@ -0,0 +1,2 @@
+.root {
+}
```
Comment (IhorBalan) [inline src/modules/core/components/phone-field/phone-field.module.scss:2]: remove this file

File: package.json [hunk 2]
```diff
@@ -70,6 +70,7 @@
     "react-error-boundary": "4.0.13",
     "react-grid-gallery": "1.0.0",
     "react-hook-form": "7.51.5",
+    "react-international-phone": "^4.3.0",
```
Comment (IhorBalan) [inline package.json:73]: better use fixed version "4.3.0"

File: src/modules/core/components/dropdown-menu/dropdown-menu.tsx [hunk 3]
```diff
@@ -61,9 +62,16 @@ export function DropdownMenu<OptionData = undefined>({
                 render(item)
               ) : (
                 <>
-                  {item.icon && (
-                    <Icon name={item.icon} width={20} height={20} />
-                  )}
+                  {item.icon &&
+                    (typeof item.icon === 'string' ? (
```
Comment (IhorBalan) [inline src/modules/core/components/dropdown-menu/dropdown-menu.tsx:66]: I suggest you to change it so we will have to always pass the `icon` prop as `ReactNode`. it this case we won't need this validation. Another solution would be change the naming, for example make `icon` prop represent the iconName of the Icon and create different prop `startdornmen` to add items as `ReactNode`

File: src/modules/core/components/phone-field/phone-field.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        }
+      },
+    });
+
+  // exclude rusia, Belarus, Iran,  and Iraq.
+  const countries = defaultCountries.filter((terorist) => {
+    const { iso2 } = parseCountry(terorist);
+
+    return iso2 !== 'ru' && iso2 !== 'ir' && iso2 !== 'by' && iso2 !== 'iq';
+  });
```
Comment (IhorBalan) [inline src/modules/core/components/phone-field/phone-field.tsx:41]: Move it to the helper, we might need to reuse this

**Review by VadimTarasenko — APPROVED** [review]

**Pull request merged**